core/src/main/java/hudson/security/UnwrapSecurityExceptionFilter.java

```diff
@@ -51,7 +51,14 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
             chain.doFilter(request,response);
         } catch (ServletException e) {
             Throwable t = e.getRootCause();
-            if (t instanceof JellyTagException) {
+            while (t != null && !(t instanceof JellyTagException)) {
+                if (t instanceof ServletException) {
+                    t = ((ServletException) t).getRootCause();
+                } else {
+                    t = t.getCause();
+                }
```

Comment:
possibly add this for safety to prevent an infinite loop, wasn't sure about adding it in the while loop condition as he may want it for the first exception, and do-while wouldn't help for the same reason, also not sure if this is a valid case, but safety won't hurt here ?
```suggestion
                }
                if (e == e.getCause()) {
                 break;
                }
```